### PR TITLE
Get OCP credentials from AWS during RHEL release [DI-78]

### PR DESCRIPTION
**Pull request by JackPGreen**
Currently, credentials are duplicated between:
- AWS Secrets Manager
- GitHub Secrets

In the past this has lead to build failures when credentials were updated in only one place.

This PR updates any credentials which are duplicated, migrating them to use the value directly from AWS Secrets Manager. The methodology on this logic is described [here](https://hazelcast.atlassian.net/browse/DI-78?focusedCommentId=104219).

I'm confident [from my testing](https://hazelcast.atlassian.net/browse/DI-78?focusedCommentId=104219) the values will resolve correctly, but as the job is triggered on `tag`, I am unable to test in a non-destructive way.

I've also rescoped `secrets.HZ_ENTERPRISE_LICENSE` to reduce the cognitive complexity of looking through the actions' environment secrets.

Fixes: [DI-78](https://hazelcast.atlassian.net/browse/DI-78)

Post-merge actions:
- [ ] backport
- [ ] delete secrets from repo

[DI-78]: https://hazelcast.atlassian.net/browse/DI-78?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/tag_image_push_rhel.yml [hunk 1]
```diff
@@ -126,6 +122,14 @@ jobs:
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: 'us-east-1'
 
+      - name: Get Secrets
+        uses: aws-actions/aws-secretsmanager-get-secrets@v2
+        with:
+          secret-ids: |
+            OCP_LOGIN_USERNAME,CN/OCP_USERNAME
```
Comment (nishaatr) [inline .github/workflows/tag_image_push_rhel.yml:129]: Do we need to mask ENVs as per https://hazelcast.atlassian.net/browse/DI-133?focusedCommentId=104227?

**Review by nishaatr — CHANGES_REQUESTED** [review]

File: .github/workflows/tag_image_push_rhel.yml (hunk 1 above)
Comment (JackPGreen) [reply .github/workflows/tag_image_push_rhel.yml:129]: They are already masked by the action

See https://github.com/hazelcast/hazelcast-docker/actions/runs/11479172818/job/31944918031 where I try to print or debug log them and fail.

Comment (nishaatr) [reply .github/workflows/tag_image_push_rhel.yml:129]: great. thanks for confirming.

**Review by nishaatr — APPROVED** [review]

File: .github/workflows/tag_image_push_rhel.yml (hunk 1 above)
Comment (nishaatr) [reply .github/workflows/tag_image_push_rhel.yml:129]: I am sure you will do this but delete secrets in repo after merge

Comment (JackPGreen) [reply .github/workflows/tag_image_push_rhel.yml:129]: > I am sure you will do this but delete secrets in repo after merge

added to description.

**Review by ldziedziul — COMMENTED** [review]
> I'm confident [from my testing](https://hazelcast.atlassian.net/browse/DI-78?focusedCommentId=104219) the values will resolve correctly, but as the job is triggered on `tag`, I am unable to test in a non-destructive way.

I know it stretches the scope of work but maybe you could add DRY_RUN flag to `tag_image_push_rhel.yml`, like here: https://github.com/hazelcast/hazelcast-docker/pull/808/commits/d41b2c79f4ad560a03b0c6f31737c92c8e73b37d (along with typo fix https://github.com/hazelcast/hazelcast-docker/pull/808/commits/1560d9bd168e9a9015d0315cd39469737010ac23)

File: .github/workflows/tag_image_push_rhel.yml (hunk 1 above)
Comment (ldziedziul) [reply .github/workflows/tag_image_push_rhel.yml:120]: > I am sure you will do this but delete secrets in repo after merge

What about maintenance branches? They need these secretes as well

**Review by ldziedziul — APPROVED** [review]
Before the merge, could you dry-run it after merging https://github.com/hazelcast/hazelcast-docker/pull/813?

File: .github/workflows/tag_image_push_rhel.yml (hunk 1 above)
Comment (JackPGreen) [reply .github/workflows/tag_image_push_rhel.yml:120]: > > I am sure you will do this but delete secrets in repo after merge
> 
> What about maintenance branches? They need these secretes as well

Once the PR is backported to all maintenance branches, they wont

**Pull request merged**